File: InternationalWagesManager.Models/SalaryComponents.cs
namespace InternationalWagesManager.Models
{
    public class SalaryComponents : BaseClass
    {
        public Employee Employee { get; set; }
        public int EmployeeId { get; set; }
        public DateTime Date { get; set; }
        public float TotalHours { get; set; }
        public float? BonusHours { get; set; }
        public float? BonusWage { get; set; }
        public float? Expenses { get; set; }

    }
}

File: AngularClient/src/app/dto/salaryComponents.ts
import { Employee } from './employee';
export class SalaryComponents{
    constructor(public id?:number, public employeeId?:number,
        public date?:Date, public totalHours?:number, public bonusHours?:number, public bonusWage?:number,
        public expenses?:number ){}
}